### PR TITLE
[repository-azure] Update to the latest Azure Storage SDK v12, remove privileged runnable wrapper in favor of access helper

### PR DESCRIPTION
**Pull request by reta**
Signed-off-by: Andriy Redko <andriy.redko@aiven.io>

### Description
The fix for [1] has finally arrived to latest Azure Storage SDK releases. We could get rid of privileged runnable wrapper and use access helper (for `JacksonAdapter` serialization / deserialization). More details in [2], [3] 
 
[1] https://github.com/Azure/azure-sdk-for-java/issues/24373
[2] https://github.com/Azure/azure-sdk-for-java/pull/25004
[3] https://github.com/Azure/azure-sdk-for-java/pull/24374

### Issues Resolved
N/A
 
### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: plugins/repository-azure/src/main/java/org/opensearch/repositories/azure/AzureStorageService.java [hunk 1]
```diff
@@ -366,14 +373,7 @@ public EventLoopGroup getEventLoopGroup() {
...
         public Thread newThread(Runnable r) {
-            // See please: https://github.com/Azure/azure-sdk-for-java/pull/24374
-            final Runnable priviledged = () -> {
-                AccessController.doPrivileged((PrivilegedAction<?>) () -> {
-                    r.run();
-                    return null;
-                });
-            };
-            final Thread t = new Thread(group, priviledged, namePrefix + threadNumber.getAndIncrement(), 0);
+            final Thread t = new Thread(group, r, namePrefix + threadNumber.getAndIncrement(), 0);
```
Comment (reta) [inline plugins/repository-azure/src/main/java/org/opensearch/repositories/azure/AzureStorageService.java:376]: No need to wrap anymore :-)

Comment (saratvemulapalli) [reply plugins/repository-azure/src/main/java/org/opensearch/repositories/azure/AzureStorageService.java:376]: Nice!

**Review by saratvemulapalli — APPROVED** [review]
Thanks @reta , changes look good to me!

**Comment by saratvemulapalli** [comment]
@reta could you also backport this PR to 1.x?

**Comment by reta** [comment]
@saratvemulapalli definitely, on it

**Pull request merged**